pkg/kubevirt/machine.go

```diff
@@ -63,6 +66,7 @@ func NewMachine(ctx *context.MachineContext, client client.Client, namespace str
 		vmiInstance:        nil,
 		vmInstance:         nil,
 		sshKeys:            sshKeys,
+		dataVolume:         nil,
```

Comment:
I think this needs to be a slice instead of a single varable. It's possible for a  VM to have mulitple DataVolumeTemplates associated with it. In the use case that you and I are most familiar with (hypershift) this will not be the case right now, but it's possible in the generic capk use case.